mobile/native/src/max_quantity.rs

```diff
@@ -40,7 +39,21 @@ pub fn max_quantity(
 
     let max_coordinator_balance =
         Amount::from_sat(channel_trade_constraints.max_counterparty_balance_sats);
-    let max_trader_balance = Amount::from_sat(channel_trade_constraints.max_local_balance_sats);
+
+    // If the trader has a channel, his max balance is the channel balance and we continue,
+    // otherwise we can return here as the max amount to trade depends on what the coordinator can
+    // provide
+    let max_trader_balance = if channel_trade_constraints.is_channel_balance {
+        Amount::from_sat(channel_trade_constraints.max_local_balance_sats)
+    } else {
+        return Ok(Decimal::from_f32(calculations::calculate_quantity(
```

Comment:
🤔 what if the user is funding with the 10101 wallet?

Reply:
His max quantity still depends on the coordinator's quantity. We have a frontend validation which does not let him chose to fund with his internal wallet if the selected channel size is too big.